File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
         stack<char>st;
         if(s.size() == 1){
            if(s[0] == '}' || s[0] == ')' || s[0] == ']') return false;
         }
        for(int i = 0; i<s.size() ; i++){
            if(s[i] == '('){
                st.push(s[i]);  
            }
            if(s[i] == '['){
                st.push(s[i]);  
            }
             if(s[i] == '{'){
                st.push(s[i]);  
            }
            if(s[i] == ')'){
                  if(st.empty()) return false;
                if(st.top() != '('){
                    return false;
                    break;
                }
                st.pop();
            }
            if(s[i] == ']'){
                if(st.empty()) return false;
                if(st.top() != '['){
                    return false;
                    break;
                }
                st.pop();
            }
            if(s[i] == '}'){
                  if(st.empty()) return false;
                if(st.top() != '{'){
                    return false;
                    break;
                }
                st.pop();
            }
        }
        return st.empty();
        
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = deque()

        open_list = ["(" , "{" , "[" ]
        close_list = [")" , "}" , "]"]

        for i in range(len(s)):
            if not st or s[i] in open_list:
                st.append(s[i])
            else:
                close = st.pop()

                if (close == "(" and s[i] == ")") or  (close == "{" and s[i] == "}") or  (close == "[" and s[i] == "]"):
                    pass
                else:
                    return False
        
        if st :
            return False
        
        return True

        